packages/eventual-send/test/test-thenable.js

```diff
@@ -3,22 +3,76 @@ import { test } from './prepare-test-env-ava.js';
 
 import { E, HandledPromise } from './get-hp.js';
 
-test('E.resolve is always asynchronous', async t => {
+test('E.resolve does not coerce simple promise', async t => {
+  const p = new Promise(_ => {});
+  const p2 = E.resolve(p);
+  t.is(p, p2, 'simple promise should not coerce');
+});
+
+test('HandledPromise.resolve does not coerce simple promise', async t => {
```

Comment:
Should we be handling these tests [macro style](https://github.com/avajs/ava/blob/main/docs/01-writing-tests.md#reusing-test-logic-through-macros)?
```js
const verifySimplePromise = async (t, resolve) => {
  const p = new Promise(_ => {});
  const p2 = resolve(p);
  t.is(p, p2, 'simple promise should not coerce');
};
test('E.resolve does not coerce simple promise', verifySimplePromise, E.resolve);
test('HandledPromise.resolve does not coerce simple promise', verifySimplePromise,
  HandledPromise.resolve);

const verifyThenAttack = async (t, resolve) => {
  const p = new Promise(_ => {});
  // Work around the override mistake on SES.
  Object.defineProperty(p, 'then', { value: (res, _rej) => res('done') });
  let happened = false;
  const p2 = resolve(p).then(ret => (happened = ret));
  t.not(p, p2, 'an own "then" should cause coercion');
  t.is(happened, false, `p2 is not yet resolved`);
  t.is(await p2, 'done', `p2 is resolved`);
};
test('E.resolve protects against "then" attack', verifyThenAttack, E.resolve);
test('HandledPromise.resolve protects against "then" attack', verifyThenAttack,
  HandledPromise.resolve);

const verifyConstructorAttack = async (t, resolve) => {
  const p = new Promise(_ => {});
  let happened = false;
  // Work around the override mistake on SES.
  Object.defineProperty(p, 'constructor', {
    get() {
      happened = true;
      return Promise;
    },
  });
  const p2 = resolve(p);
  t.not(p, p2, 'an own "constructor" should cause coercion');
  // This first 'true' demonstrates our remaining vulnerability to reentrancy.
  // But the fact that p coerced to a fresh p2 means that p2 cannot
  // cause a reentrancy attack
  t.is(happened, true, `same turn`);
  await null;
  t.is(happened, true, `later turn with p still unresolved`);
};
test('E.resolve protects against "constructor" attack', verifyConstructorAttack, E.resolve);
test('HandledPromise.resolve protects against "constructor" attack', verifyConstructorAttack,
  HandledPromise.resolve);
```

Reply:
Done. Started with exactly your code.

@turadg , this code is using macro style and the unpatched ses-ava. Is it surprising that it seems to work?

Reply:
Not surprising. It would be if you used the `test.macro` helper because ses-ava has a whiltelist of methods and `macro` isn't in it https://github.com/endojs/endo/blob/235931f48757fc8c2ba472a7e7c5f46fca6286fe/packages/ses-ava/src/ses-ava-test.js#L49-L58